src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -264,9 +269,13 @@ protected void flush() {
   protected Object readProtocolWithCheckingBroken() {
     try {
       return Protocol.read(inputStream);
-    } catch (JedisConnectionException exc) {
-      broken = true;
-      throw exc;
+    } catch (RuntimeException e) {
+      if (connBrokenDeterminer.determine(e)) {
+        broken = true;
+        throw new JedisConnectionException(e);
```

Comment:
`e` is likely to be JedisConnectionException, at least by default. So why not just throw `e` without wrapping?

Reply:
There's a chance for `e` to be `JedisDataException`  like #1087, or even `ClassCastException`.

Reply:
**At this line, by default,** there is no chance `e` would be `JedisDataException` or `ClassCastException`.

By default, `ConnectionBrokenDeterminer` has only `JedisConnectionExceptionAsConnectionBroken` which just checks if provided `RuntimeException` is an instance of `JedisConnectionException`. So it is either `JedisConnectionException` or `JedisNoReachableClusterNodeException`. So, if the condition of `if` is true, for the default `ConnectionBrokenDeterminer`, `e` is `JedisConnectionException` (unless it is cluster mode and every node in cluster is down). And wrapping a most likely `JedisConnectionException` within another `JedisConnectionException` doesn't seem to be a good idea to me.

Reply:
OK then does it make sense to wrap `e` with JedisConnectionException only when `e` is not an instance of JedisConnectionException? 
Also you may want to be aware that ConnectionBrokenDeterminer is designed as a plug-in, and the purpose of ConnectionBrokenDeterminer is to classify some non-JedisConnectionExceptions as a JedisConnectionException.

Reply:
Oh! The way I saw is that ConnectionBrokenDeterminer is to help defining the connection as broken. Which, to simply put, is to set the variable `broken = true`. Our ideas are somehow related, but seem to be different.

How is wrapping other exception as JedisConnectionException helpful? What if we don't?

Reply:
Users could also have try-catch statement in their codebase, and expect connection related exception to be thrown as JedisConnectionException. We have reported too many issues for ClassCastException in Jedis codebase (users may feel really odd and strange) which was due to connection messed up (so the response of Redis is not expected type), and throwing CCE as it is doesn't help for users.

Reply:
Understood. Thanks!